add dbt.current_timestamp() (#4741)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/cross-database-macros.md

    - [type\_numeric](#type_numeric)
    - [type\_string](#type_string)
    - [type\_timestamp](#type_timestamp)

>     - [dbt.current\_timestamp()](#dbtcurrent_timestamp)

Reviewer: We should be consistent by either using the `dbt.` prefix for all of these macros or none.

All the cross-database macros may optionally use the `dbt.` namespace prefix. It is optional because the `dbt` namespace is the default one.

```suggestion
    - [current\_timestamp](#current_timestamp)
```

    - [type\_numeric](#type_numeric)
    - [type\_string](#type_string)
    - [type\_timestamp](#type_timestamp)

>     - [dbt.current\_timestamp()](#dbtcurrent_timestamp)

Reviewer: ```suggestion
    - [current\_timestamp](#current_timestamp)
```

TIMESTAMP
```

### current_timestamp

This macro returns the current date and time for the system. Depending on the adapter:

- The result may be an aware or naive timestamp.
- The result may correspond to the start of the statement or the start of the transaction.

> Refer to your adapter's documentation for specific timestamp information.

Author: ```suggestion
```

TIMESTAMP
```

> ### dbt.current_timestamp()

Reviewer: ```suggestion
### current_timestamp
```

> The `dbt.current_timestamp()` macro allows you to generate the current timestamp in a format suitable for database queries. This is useful when you need to capture the current date and time within your SQL queries or models.

Author: @dbeatty10 is this the same for our other adapters? starburst, fabric, spark, databricks?

Reviewer: The thing that is true across all adapters:
- This macro returns the current date and time for the system.

The things that may vary:
- whether the result is an aware or naive timestamp
- whether it corresponds to the start of the statement or the start of the transaction (or something else)

> The `dbt.current_timestamp()` macro allows you to generate the current timestamp in a format suitable for database queries. This is useful when you need to capture the current date and time within your SQL queries or models.

Reviewer: Here's what I'd suggest based off these docs:
- [postgres `now()`](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-CURRENT)
- [redshift `getdate()`](https://docs.aws.amazon.com/redshift/latest/dg/r_GETDATE.html)
- [bigquery `current_timestamp()`](https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#current_timestamp)
- [snowflake `current_timestamp()`](https://docs.snowflake.com/en/sql-reference/functions/current_timestamp)

```suggestion
This macro returns the current date and time for the system. Depending on the adapter:
- The result may be an aware or naive timestamp.
- The result may correspond to the start of the statement or the start of the transaction.
```

Or alternatively:

```suggestion
This macro returns the current date and time for the system. The semantics will vary depending on the adapter.
```

**Args**
- None

**Usage**
- You can use the `dbt.current_timestamp()` macro within your dbt SQL files like this:

```sql
{{ dbt.current_timestamp() }}
```

<Tabs>
<TabItem value="postgres" label="Sample output (PostgreSQL)">

```sql

> CURRENT_TIMESTAMP OR TIMESTAMP?

Author: not sure which is the right output?

Reviewer: I added suggestions for each adapter.

Here's how I determined the output:

1. Create an analyses file:

    `analyses/current_timestamp.sql `
    
    ```sql
    {{ dbt.current_timestamp() }}
    ```

2. Compile it and look at the output:

    ```shell
    dbt compile --profile postgres
    ```
    
    Look at the compiled output in `target/compiled/my_project/analyses/current_timestamp.sql`.

> CURRENT_TIMESTAMP OR TIMESTAMP?

Reviewer: ```suggestion
now()
```

```
</TabItem>

> <TabItem value="sonwflake" label="Sample output (Snowflake)">

Reviewer: ```suggestion
<TabItem value="snowflake" label="Sample output (Snowflake)">
```

```sql

> CURRENT_TIMESTAMP() OR [TIMESTAMP_TZ](https://docs.snowflake.com/en/sql-reference/data-types-datetime#timestamp-ltz-timestamp-ntz-timestamp-tz)?

Reviewer: ```suggestion
convert_timezone('UTC', current_timestamp())
```

```
</TabItem>

<TabItem value="bq" label="Sample output (BigQuery)">

```sql

> CURRENT_DATETIME() OR TIMESTAMP?

Reviewer: ```suggestion
current_timestamp()
```

```
</TabItem>

<TabItem value="redshift" label="Sample output (Redshift)">

```sql

> CURRENT_TIMESTAMP OR TIMESTAMP?

Reviewer: ```suggestion
getdate()
```

TIMESTAMP
```

### current_timestamp

This macro returns the current date and time for the system. Depending on the adapter:

- The result may be an aware or naive timestamp.
- The result may correspond to the start of the statement or the start of the transaction.

> **Args**

Author: @dbeatty10 added this bit

TIMESTAMP
```

### current_timestamp

This macro returns the current date and time for the system. Depending on the adapter:

- The result may be an aware or naive timestamp.
- The result may correspond to the start of the statement or the start of the transaction.

> Refer to your adapter's documentation for specific timestamp information.

Reviewer: I'm not aware of any adapter-specific documentation including information about timestamps.

So it seems like we should either:
a) open issues to add timestamp-specific info to each adapter, or
b) remove this line

Do you have a preference?

> Refer to the documentation for your adapter for specific timestamp information.

Author: @dbeatty10 added this bit

> Refer to the documentation for your adapter for specific timestamp information.

Author: ```suggestion
Refer to your adapter's documentation for specific timestamp information.
```

**Args**
- None

**Usage**
- You can use the `current_timestamp()` macro within your dbt SQL files like this:

```sql
{{ dbt.current_timestamp() }}
```

<Tabs>
<TabItem value="bq" label="Sample output (BigQuery)">

```sql
current_timestamp()
```
</TabItem>

<TabItem value="postgres" label="Sample output (PostgreSQL)">

```sql
now()
```

> </TabItem>

Reviewer: All the other sample output on the page _only_ includes Postgres. I think the idea there was so that we didn't need to keep ~30 macros x ~5 adapters = 150 different items up-to-date.

So it seems like we should either:
a) open issues to add sample output for each macro for each adapter, or
b) remove the sample output for `current_timestamp` for BQ, Snowflake, Redshift

We could also choose to include examples across adapters for _only_ the `current_timestamp` macro, but that would be discordant.

My strong preference would be option b) in order to make things more simple and easier to keep up-to-date.

What do you think @mirnawong1 ?

Author: oh ok! i saw this issue https://github.com/dbt-labs/dbt-utils/pull/597 and for some reason thought to add the other adapters!

Author: let's keep it with postgres so it's a simple example!
